straxen/plugins/led_cal/led_calibration.py

```diff
@@ -122,48 +178,130 @@ def get_records(raw_records, baseline_window):
     mask = np.where((records["record_i"] == 0) & (records["length"] == 160))[0]
     records = records[mask]
     bl = records["data"][:, baseline_window[0] : baseline_window[1]].mean(axis=1)
+    rms = records["data"][:, baseline_window[0] : baseline_window[1]].std(axis=1)
     records["data"][:, :160] = -1.0 * (records["data"][:, :160].transpose() - bl[:]).transpose()
+    records["baseline"] = bl
+    records["baseline_rms"] = rms
     return records
 
 
+def get_led_windows(
+    records,
+    default_window,
+    led_hit_extension,
+    hit_min_amplitude,
+    hit_min_height_over_noise,
+):
+    """Search for hits in the records, if a hit is found, return an interval around the hit given by
+    led_hit_extension. If no hit is found in the record, return the default window.
+
+    :param records: Array of the records to search for LED hits.
+    :param default_window: Default window to use if no LED hit is found given as a tuple (start,
+        end)
+    :param led_hit_extension: The integration window around the first hit found to use. A tuple of
+        form (samples_before, samples_after) the first LED hit.
+    :param hit_min_amplitude: Minimum amplitude of the signal to be considered a hit.
+    :param hit_min_height_over_noise: Minimum height of the signal over noise to be considered a
+        hit. :return (len(records), 2) array: Integration window for each record
+
+    """
+    hits = strax.find_hits(
```

Comment:
What is the lowest hits detectable? 15 ADC? Can this bias the result?

Reply:
I didn't know how to test the plugin in place in straxen earlier, so I copied the function and run it outside. However doing that I did not have access to the URL config resources. I tested it again in place and it turns out with the hard cutoff of 15 ADC I see a hard cut in the amplitude spectra at 15 ADC. I removed the hard cutoff and always use the `hit_min_height_over_noise` which removes the hard cut in the spectra.